### ISSUE TITLE
Installing dev dependencies of path dependencies

### ISSUE THREAD

**Issue by Apakottur** [python-poetry/poetry#5748]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

## Feature Request
Consider the following setup, where one package (`core`) is using another (`shared`), and both have deps and dev-deps.

```toml
# pyproject.toml for `core`
[tool.poetry.dependencies]
package_1 = "1.0.0"
shared = { path = "shared", develop = true }

[tool.poetry.dev-dependencies]
package_2 = "1.0.0"
```

```toml
# pyproject.toml for `shared`
[tool.poetry.dependencies]
package_3 = "1.0.0"

[tool.poetry.dev-dependencies]
package_4 = "1.0.0"
```

If I go to `core` and run `poetry install` then I will install `core, api, package_1, package_2` and `package_3`, but not `package_4`.
Is there a way to also include dev-dependencies of path style dependencies?

In other words, is there a way to include `package_4` when installing `core`?

**Comment by dimbleby** [comment]
No, poetry won't install the dev dependencies of dependencies.  

This is nothing to do with path dependencies, B's dev dependencies simply aren't supposed to be interesting to A - that's why they were declared as dev dependencies.

If - as is often the case eg for `flake8`, `black`, etc - projects happen to have the same dev dependencies: well each project declares what it needs.  

Just declare that `package_4` is a dev dependency of `core`.

**Comment by finswimmer** [comment]
Hello @Apakottur,

it's exactly as @dimbleby said."dev-dependencies" are dependencies only needed during development. If you need a "dev-depdency" of one of your deps, it's obviously not needed only during development and should be declared as a regular dependency.

fin swimmer

**Comment by Apakottur** [comment]
Thanks for the clarification.
I ended up using the `extras` feature to specify the dev dependencies of the internal package, which then lets me access them in the core one when necessary.
